pkg/geo/geomfn/subdivide.go

```diff
@@ -0,0 +1,364 @@
+	if err != nil {
+		return errors.Newf("error clipping geom.T: %v", err)
+	}
+	clipped2, err := clipGeomTByBoundingBox(gt, &subBox2)
+	if err != nil {
+		return errors.Newf("error clipping geom.T: %v", err)
+	}
+	depth++
+	err = subdivideRecursive(clipped1, maxVertices, depth, col, dim)
+	if err != nil {
+		return err
+	}
+	err = subdivideRecursive(clipped2, maxVertices, depth, col, dim)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// isGeomTCollection returns whether the given geom.T is of a collection type (wrapper for a geometry function).
+func isGeomTCollection(gt geom.T) (bool, error) {
+	g, err := geo.MakeGeometryFromGeomT(gt)
+	if err != nil {
+		return false, err
+	}
+	return IsCollection(g)
+}
+
+// forEachGeomTInCollection calls the provided function for each geometry in the passed in collection.
+// Ignores MultiPoint input.
+func forEachGeomTInCollection(gt geom.T, fn func(geom.T) error) error {
+	switch gt := gt.(type) {
+	case *geom.GeometryCollection:
+		for i := 0; i < gt.NumGeoms(); i++ {
+			err := fn(gt.Geom(i))
+			if err != nil {
+				return err
+			}
+		}
+		return nil
+	case *geom.MultiPolygon:
+		for i := 0; i < gt.NumPolygons(); i++ {
+			err := fn(gt.Polygon(i))
+			if err != nil {
+				return err
+			}
+		}
+	case *geom.MultiLineString:
+		for i := 0; i < gt.NumLineStrings(); i++ {
+			err := fn(gt.LineString(i))
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func calculateSplitPoint(
+	gt geom.T, nVertices int, splitHorizontally bool, clip *geo.CartesianBoundingBox,
+) (float64, error) {
+	var pivot float64
+	switch gt := gt.(type) {
+	case *geom.Polygon:
+		var err error
+		pivot, err = findPolygonsMostCentralPointValue(gt, nVertices, splitHorizontally, clip)
+		if err != nil {
+			return 0, errors.Newf("error finding most central point for polygon: %v", err)
+		}
+		if splitHorizontally {
+			// Ignore point on the boundaries
+			if clip.LoX == pivot || clip.HiX == pivot {
+				pivot = (clip.LoX + clip.HiX) / 2
+			}
+		} else {
+			// Ignore point on the boundaries
+			if clip.LoY == pivot || clip.HiY == pivot {
+				pivot = (clip.LoY + clip.HiY) / 2
+			}
+		}
+	default:
+		if splitHorizontally {
+			pivot = (clip.LoX + clip.HiX) / 2
+		} else {
+			pivot = (clip.LoY + clip.HiY) / 2
+		}
+	}
+	return pivot, nil
+}
+
+// findPolygonsMostCentralPointValue finds the value of the most central point for a geom.Polygon.
+// Can be X or Y value, depending on the splitHorizontally argument.
+func findPolygonsMostCentralPointValue(
+	poly *geom.Polygon, nVertices int, splitHorizontally bool, clip *geo.CartesianBoundingBox,
+) (float64, error) {
+	pivot := math.MaxFloat64
+	ringToTrimIndex := 0
+	ringArea := float64(0)
+	pivotEps := math.MaxFloat64
+	var ptEps float64
+	nVerticesOuterRing, err := countVertices(poly.LinearRing(0))
+	if err != nil {
+		return 0, errors.Newf("error counting vertices: %v", err)
+	}
+	// if there are more points in holes than in outer ring
```

Comment:
nit: this feels like an incomplete sentence

Reply:
Fixed